src/relay/qnn/op/convolution.cc

```diff
@@ -57,22 +59,27 @@ bool QnnConv2DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   ICHECK(param->out_dtype.bits() > 0) << "Output dtype bits should be greater than 0.";
 
   // Check the types of scale and zero points.
+  for (size_t i = 2; i < 5; ++i) {
```

Comment:
Am I counting wrong or is this skipping `weight_scale`?

Reply:
Weight scale is Assigned further down:
` AssignType(types[5], DataType::Float(32), weight->shape[axis], reporter);  // weight_scale`

It's one of those times where I assume it should be an input, but it's getting set here.

Reply:
I think @mbrookhart added check for those types checked by `ICHECK(IsScalarType(...))` below. There is no check for weight scale being scalar because weight scale can be a vector for per channel case.